gtsfm/utils/homography_decomposition.py

```diff
@@ -0,0 +1,384 @@
+"""
+Decompose an homography matrix into the possible rotations, translations,
+and plane normal vectors.
+
+Based off of homography decomposition implementation in COLMAP:
+    https://github.com/colmap/colmap/blob/dev/src/base/homography_matrix.cc
+    https://github.com/colmap/colmap/blob/dev/src/base/homography_matrix.h
+See how it is used in COLMAP:
+https://github.com/colmap/colmap/blob/dev/src/estimators/two_view_geometry.cc#L198
+
+COLMAP and OpenCV's implementationsare based off of:
+Ezio Malis, Manuel Vargas, and others. Deeper understanding of the homography decomposition for vision-based control. 2007.
+https://hal.inria.fr/inria-00174036/PDF/RR-6303.pdf
+
+OpenCV does not support the case of intrinsics from two separate cameras, however.
+
+Authors: John Lambert (Python), from original C++
+"""
+
+from typing import List, Tuple
+
+import numpy as np
+from gtsam import Rot3, Unit3
+
+"""
```

Comment:
I don't like having code examples when we have linked the source. For brevity as well as potential copyright issues.